### PR TITLE
v1.4.0 dependency updates

### LINKED ISSUE #222: Example should be modify to use callback refs instead of deprecated string ref
Hello,

This is a really good library.

However, the examples could be improved by using callback refs instead of string refs.
Maybe also explain in the README how to call the runAnimation method (because finding it in the example can be tricky for some).

For references: https://facebook.github.io/react/docs/refs-and-the-dom.html, and specially at https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs

> If you worked with React before, you might be familiar with an older API where the ref attribute is a string, like "textInput", and the DOM node is accessed as this.refs.textInput. We advise against it because string refs have some issues, are considered legacy, and are likely to be removed in one of the future releases. If you're currently using this.refs.textInput to access refs, we recommend the callback pattern instead.

Regards

### LINKED ISSUE #224: upgrade to react transition group v2 dependency

### LINKED ISSUE #229: Support React 16
Could you please update the package.json file with newer version of `react-transition-group` ?
The version 1.2.1 supports React 16.

UPDATE
Noticed that React 16 is supported in `master` branch, so we just need a new release (tag).

### LINKED ISSUE #235: Old Modules
Hey guys. Some of modules cannot work any more. Is there any other ways to take a look at your demos?

### LINKED ISSUE #236: Remove usages of unsafe lifecycle methods
React team recently published a blogpost: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

TL;DR: `cWM`, `cWRP`, and `cWU` methods are considered unsafe for future async rendering and will be deprecated soon. The recommended way for open source maintainers to deal with it is to use [`react-lifecycles-compat`](https://github.com/reactjs/react-lifecycles-compat) package and migrate to new `static getDerivedStateToProps` method

### LINKED ISSUE #237: Lodash version is too low
https://snyk.io/vuln/npm:lodash:20180130

Now velocity-react uses lodash 3.10.1, but it has security hole. The latest version 4.17.5 fixes this issue.

### PR DESCRIPTION
**Pull request by finneganh**
- Changes our `componentWillUpdate` call to `componentDidUpdate` to eliminate a
   React 16.3 StrictMode warning
 - Migration to [react-transition-group](https://github.com/reactjs/react-transition-group) v2
 - Switches from lodash 3 to lodash 4
 - Changes demo to use React 16.3

Also, removes `react-tween-state` because it was contributing to a
potential memory leak in the demo.

Fixes #237, fixes #236, fixes #235, fixes #229, fixes #222, fixes #224

Closes #233

### CODE REVIEW

**Comment by finneganh** [comment]
Prettier stepped in on the demo files. Sorry for the big diff.

**Comment by finneganh** [comment]
Also, how you been, @tonybaroneee?

**Comment by tonybaroneee** [comment]
No worries 😄This is cool!

And can't complain, how about you @finneganh?

**Comment by tonybaroneee** [comment]
Also, I've been meaning to check out Prettier :)
